integration-cli/docker_cli_inspect_test.go

```diff
@@ -176,57 +161,48 @@ func (s *DockerSuite) TestInspectContainerFilterInt(c *check.C) {
 	runCmd := exec.Command(dockerBinary, "run", "-i", "-a", "stdin", "busybox", "cat")
 	runCmd.Stdin = strings.NewReader("blahblah")
 	out, _, _, err := runCommandWithStdoutStderr(runCmd)
-	if err != nil {
-		c.Fatalf("failed to run container: %v, output: %q", err, out)
-	}
+	c.Assert(err, checker.IsNil, check.Commentf("failed to run container: %v, output: %q", err, out))
 
 	id := strings.TrimSpace(out)
 
 	out, err = inspectField(id, "State.ExitCode")
-	c.Assert(err, check.IsNil)
+	c.Assert(err, checker.IsNil)
 
 	exitCode, err := strconv.Atoi(out)
-	if err != nil {
-		c.Fatalf("failed to inspect exitcode of the container: %s, %v", out, err)
-	}
+	c.Assert(err, checker.IsNil, check.Commentf("failed to inspect exitcode of the container: %s, %v", out, err))
 
 	//now get the exit code to verify
 	formatStr := fmt.Sprintf("--format='{{eq .State.ExitCode %d}}'", exitCode)
 	out, _ = dockerCmd(c, "inspect", formatStr, id)
-	if result, err := strconv.ParseBool(strings.TrimSuffix(out, "\n")); err != nil || !result {
-		c.Fatalf("Expected exitcode: %d for container: %s", exitCode, id)
-	}
+	result, err := strconv.ParseBool(strings.TrimSuffix(out, "\n"))
+	c.Assert(err, checker.Equals, nil)
+	c.Assert(result, checker.Equals, true)
 }
 
 func (s *DockerSuite) TestInspectImageGraphDriver(c *check.C) {
 	testRequires(c, DaemonIsLinux)
 	imageTest := "emptyfs"
 	name, err := inspectField(imageTest, "GraphDriver.Name")
-	c.Assert(err, check.IsNil)
+	c.Assert(err, checker.IsNil)
 
-	if name != "devicemapper" && name != "overlay" && name != "vfs" && name != "zfs" && name != "btrfs" && name != "aufs" {
-		c.Fatalf("%v is not a valid graph driver name", name)
-	}
+	checkValidGraphDriver(c, name)
 
+	// requires devicemapper
 	if name != "devicemapper" {
 		return
```

Comment:
can this be turned into a `testRequires` argument some how? It seems like the test should be skipped on device mapper.